toolchain/gcc.sh

```diff
@@ -0,0 +1,205 @@
+
+# Binutils-gdb - Do in one step if possible
+if [ -e "binutils-gdb" ]; then
+  mkdir -p ${BUILDPREFIX}/binutils-gdb
+  cd ${BUILDPREFIX}/binutils-gdb
+  CFLAGS="-g -O2 -Wno-error=implicit-function-declaration" \
+  CXXFLAGS="-g -O2 -Wno-error=implicit-function-declaration" \
+  ../../binutils-gdb/configure        \
+      --target=riscv32-unknown-elf    \
+      --prefix=${INSTALLPREFIX}       \
+      --with-expat                    \
+      --with-libgmp-prefix=${SRCPREFIX}/gmp-${LIBGMP_VERS}/inst \
+      --disable-werror                \
+      ${EXTRA_OPTS}                   \
+      ${EXTRA_BINUTILS_OPTS}
+  make -j${PARALLEL_JOBS}
+  make install
+else
+  # Binutils
+  mkdir -p ${BUILDPREFIX}/binutils
+  cd ${BUILDPREFIX}/binutils
+  CFLAGS="-g -O2 -Wno-error=implicit-function-declaration" \
+  CXXFLAGS="-g -O2 -Wno-error=implicit-function-declaration" \
+  ../../binutils/configure            \
+      --target=riscv32-unknown-elf    \
+      --prefix=${INSTALLPREFIX}       \
+      --disable-werror                \
+      --disable-gdb                   \
+      ${EXTRA_OPTS}                   \
+      ${EXTRA_BINUTILS_OPTS}
+  make -j${PARALLEL_JOBS}
+  make install
+  # GDB
+  mkdir -p ${BUILDPREFIX}/gdb
+  cd ${BUILDPREFIX}/gdb
+  CFLAGS="-g -O2 -Wno-error=implicit-function-declaration" \
+  CXXFLAGS="-g -O2 -Wno-error=implicit-function-declaration" \
+  ../../gdb/configure                 \
+      --target=riscv32-unknown-elf    \
+      --prefix=${INSTALLPREFIX}       \
+      --with-expat                    \
+      --with-libgmp-prefix=${SRCPREFIX}/gmp-${LIBGMP_VERS}/inst \
+      --disable-werror                \
+      ${EXTRA_OPTS}                   \
+      ${EXTRA_BINUTILS_OPTS}
+  make -j${PARALLEL_JOBS} all-gdb
+  make install-gdb
+fi
+
+# GCC
+cd ${SRCPREFIX}/gcc
+./contrib/download_prerequisites
+mkdir -p ${BUILDPREFIX}/gcc-stage1
+cd ${BUILDPREFIX}/gcc-stage1
+../../gcc/configure                                     \
+    --target=riscv32-unknown-elf                        \
+    --prefix=${INSTALLPREFIX}                           \
+    --with-sysroot=${INSTALLPREFIX}/riscv32-unknown-elf \
+    --with-newlib                                       \
+    --without-headers                                   \
+    --disable-shared                                    \
+    --enable-languages=c                                \
+    --disable-werror                                    \
+    --disable-libatomic                                 \
+    --disable-libmudflap                                \
+    --disable-libssp                                    \
+    --disable-quadmath                                  \
+    --disable-libgomp                                   \
+    --disable-nls                                       \
+    --disable-bootstrap                                 \
+    --enable-multilib                                   \
+    --with-multilib-generator="rv32ima-ilp32--zca*zcb*zcmt*zcmb*zcmp " \
```

Comment:
@tobywenman  In the end do you have all possible combinations of Zc in install/riscv32-unkown-elf/lib ? When I compiled this a week ago, it didnt generate any Zc library and ended up removing the wild flags and that worked !